cmake/BuildLibSanitizers.cmake

```diff
@@ -23,36 +23,65 @@ macro(GEN_BUILD_TARGET name libsanitizer_path libfuzzer_path
 
   # Strip preinit object files in static libraries, otherwise a message
   # `.preinit_array section is not allowed in DSO` will prevent building DSO.
-  add_custom_target(strip_lib_${name}
-    COMMENT "Strip sanitizer library ${name}"
-    COMMAND ${CMAKE_C_COMPILER_AR} d ${libsanitizer_name} ${strip}
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-    BYPRODUCTS ${libsanitizer_name}
-    DEPENDS copy_libs_${name}
-  )
+  if(strip AND NOT "${strip}" STREQUAL "")
+    add_custom_target(strip_lib_${name}
+      COMMENT "Strip sanitizer library ${name}"
+      COMMAND ${CMAKE_C_COMPILER_AR} d ${libsanitizer_name} ${strip}
+      WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+      BYPRODUCTS ${libsanitizer_name}
+      DEPENDS copy_libs_${name}
+    )
+  else()
+    add_custom_target(strip_lib_${name}
+      COMMENT "Skip stripping sanitizer library ${name} (no objects to strip)"
+      DEPENDS copy_libs_${name}
+    )
+  endif()
 
-  add_custom_target(build_dso_${name} ALL
-    COMMENT "Build sanitizer library ${name}"
-    COMMAND ${CMAKE_C_COMPILER} -Wl,--whole-archive ${libfuzzer_name}
-       ${libsanitizer_name} -Wl,--no-whole-archive -lstdc++ -lpthread -ldl
-       -shared -o ${sanitizer_dso_name}
-    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-    BYPRODUCTS ${sanitizer_dso_name}
-    DEPENDS strip_lib_${name}
-  )
+  if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+    add_custom_target(build_dso_${name} ALL
+      COMMENT "Build sanitizer library ${name}"
+      COMMAND ${CMAKE_C_COMPILER} -Wl,-force_load,${libfuzzer_name}
+         -Wl,-force_load,${libsanitizer_name} -lstdc++ -lpthread
+         -dynamiclib -o ${sanitizer_dso_name}
+      WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+      BYPRODUCTS ${sanitizer_dso_name}
+      DEPENDS strip_lib_${name}
+    )
+  else()
+    add_custom_target(build_dso_${name} ALL
+      COMMENT "Build sanitizer library ${name}"
+      COMMAND ${CMAKE_C_COMPILER} -Wl,--whole-archive ${libfuzzer_name}
+         ${libsanitizer_name} -Wl,--no-whole-archive -lstdc++ -lpthread -ldl
+         -shared -o ${sanitizer_dso_name}
+      WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+      BYPRODUCTS ${sanitizer_dso_name}
+      DEPENDS strip_lib_${name}
+    )
+  endif()
```

Comment:
It is better to put only a variable with command under condition and use this variable in `add_custom_target()`. The same below.